### PR TITLE
Revert "Revert "CTW-555 request header""

### PR DESCRIPTION
**Pull request by melocule**
Reverts zeus-health/ctw-component-library#319

[Original PR](https://github.com/zeus-health/ctw-component-library/pull/315)

### CODE REVIEW

File: .changeset/friendly-berries-tease.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@zus-health/ctw-component-library": patch
+---
+
+Re-add header that indicates the source of outgoing requests.
```
Comment (luthfib) [inline .changeset/friendly-berries-tease.md:5]:
```suggestion
Re-add header that indicates the source of outgoing requests..
```

**Comment by luthfib** [comment]
I think @mrosata suggestion makes sense where he said we could do:

```
const { fetch: originalFetch } = window;

const {
  SERVICE_NAME,
  ZUS_HEADER_NAME,
} = process.env;

window.fetch = async (...args) => {
  let [resource, config ] = args;

  if (isZusOrigin(resource)) {
    config.headers[ZUS_HEADER_NAME] = SERVICE_NAME;
  }

  return originalFetch(resource, config);
};
```

@eric-gunther @mrosata would this get flagged at all by security teams since we are modifying the window fetch?

**Comment by eric-gunther** [comment]
> I think @mrosata suggestion makes sense where he said we could do:
> 
> ```ts
> const { fetch: originalFetch } = window;
> 
> const {
>   SERVICE_NAME,
>   ZUS_HEADER_NAME,
> } = process.env;
> 
> window.fetch = async (...args) => {
>   let [resource, config ] = args;
> 
>   if (isZusOrigin(resource)) {
>     config.headers[ZUS_HEADER_NAME] = SERVICE_NAME;
>   }
> 
>   return originalFetch(resource, config);
> };
> ```
> 
> @eric-gunther @mrosata would this get flagged at all by security teams since we are modifying the window fetch?

Couldn't this potentially break functionality in whatever app is embedding our component? I suppose it's limited to just requests that have a `resource` argument but it still seems too broad to me.

**Review by pmadria3 — COMMENTED** [review]
This looks good to me if we are keeping the original PR @luthfib @eric-gunther thoughts?

**Review by luthfib — APPROVED** [review]

**Pull request merged**